base/regex.jl

```diff
@@ -517,3 +517,49 @@ function hash(r::Regex, h::UInt)
     h = hash(r.compile_options, h)
     h = hash(r.match_options, h)
 end
+
+## String operations ##
+
+unwrap_string(r::Regex) = r.pattern
+unwrap_string(s::Union{AbstractString,AbstractChar}) = s
+
+"""
+    *(s::Regex, t::Union{Regex,AbstractString,AbstractChar}) -> Regex
+    *(s::Union{Regex,AbstractString,AbstractChar}, t::Regex) -> Regex
+
+Concatenate regexes, strings and/or characters, producing a [`Regex`](@ref).
+
+!!! compat "Julia 1.2"
+     This method requires at least Julia 1.2.
+
+# Examples
+```jldoctest
+julia> r"Hello " * "world"
+r"Hello world"
+
+julia> 'j' * r"ulia"
+r"julia"
+```
+"""
+function *(r1::Union{Regex,AbstractString,AbstractChar}, rs::Union{Regex,AbstractString,AbstractChar}...)
+    opts = unique((r.compile_options, r.match_options) for r in (r1, rs...) if r isa Regex)
+    length(opts) == 1 ||
+        throw(ArgumentError("cannot multiply regexes with incompatible options"))
```

Comment:
This could be supported with [internal option setting](https://www.pcre.org/original/doc/html/pcrepattern.html#SEC13), probably best combined with a [non-capturing subpattern](https://www.pcre.org/original/doc/html/pcrepattern.html#SEC14). So something like this:
```jl
"(?sm:$(r1.pattern))(?i:$(r2.pattern))"
```
This could work uniformly but it would be nice to avoid the extra line noise in the presumably common case where the options match. You could get even fancier and keep options that are shared by all of the regexes in the result regex's global options while only putting the options that are specific to each subpattern in it's option set.